clustering/web/infinispan/src/main/java/org/wildfly/clustering/web/infinispan/session/fine/FineSessionAttributesFactory.java

```diff
@@ -130,17 +131,23 @@ public void close() {
     private AtomicReference<Map<String, UUID>> getValue(String id, boolean purgeIfInvalid) {
         Map<String, UUID> names = this.namesCache.get(new SessionAttributeNamesKey(id));
         if (names != null) {
-            for (Map.Entry<String, UUID> nameEntry : names.entrySet()) {
-                V value = this.attributeCache.get(new SessionAttributeKey(id, nameEntry.getValue()));
+            // Validate all attributes
+            Map<SessionAttributeKey, String> attributes = new HashMap<>();
```

Comment:
Given that we know the final size of this map map we should theoretically explicitly size it to avoid unnecessary resizes – `new HashMap<>(names.entrySet().size(), 1.0f));`.